core/drivers/stm32_hpdma.c

```diff
@@ -143,41 +174,18 @@ static TEE_Result apply_rif_config(struct hpdma_pdata *hpdma_d, bool is_tdcid)
 		io_clrsetbits32(hpdma_d->base + _HPDMA_CIDCFGR(i),
 				_HPDMA_CIDCFGR_CONF_MASK,
 				hpdma_d->conf_data->cid_confs[i]);
-
-		cidcfgr = io_read32(hpdma_d->base + _HPDMA_CIDCFGR(i));
-
-		/*
-		 * Take semaphore if the resource is in semaphore
-		 * mode and secured.
-		 */
-		if (!stm32_rif_semaphore_enabled_and_ok(cidcfgr, RIF_CID1) ||
-		    !(io_read32(hpdma_d->base + _HPDMA_SECCFGR) & BIT(i))) {
-			res =
-			stm32_rif_release_semaphore(hpdma_d->base +
-						    _HPDMA_SEMCR(i),
-						    HPDMA_NB_MAX_CID_SUPPORTED);
-			if (res) {
-				EMSG("Couldn't release semaphore for res%u", i);
-				return TEE_ERROR_ACCESS_DENIED;
-			}
-		} else {
-			res =
-			stm32_rif_acquire_semaphore(hpdma_d->base +
-						    _HPDMA_SEMCR(i),
-						    HPDMA_NB_MAX_CID_SUPPORTED);
-			if (res) {
-				EMSG("Couldn't acquire semaphore for res%u", i);
-				return TEE_ERROR_ACCESS_DENIED;
-			}
-		}
 	}
 
 	/*
 	 * Lock RIF configuration if configured. This cannot be undone until
 	 * next reset.
 	 */
-	io_clrsetbits32(hpdma_d->base + _HPDMA_RCFGLOCKR, _HPDMA_RCFGLOCKR_MASK,
-			hpdma_d->conf_data->lock_conf[0]);
+	io_setbits32(hpdma_d->base + _HPDMA_RCFGLOCKR,
+		     hpdma_d->conf_data->lock_conf[0] & _HPDMA_RCFGLOCKR_MASK);
```

Comment:
`& _HPDMA_RCFGLOCKR_MASK` not needed.
Ditto for the stm32_fmc and stm32_gpio driver patches.